scripts/generating_yaml.py

```diff
@@ -0,0 +1,286 @@
+        camera_to_usb_widget, camera_into_USBhub_port=camera_into_USBhub)
+
+    def add_device_widget(add_device):
+        """Function to add the details selected using the drop-down menu widgets to the 'data_list' dictionary.
+        The function is called by the widget and is then run when the user clicks on the widget button.
+
+        Args:
+            add_device : on clicking the widget, a device set is saved to the dict.
+
+        Returns:
+            Appends the data_list dict with the information chosen from the device information widgets
+
+        """
+        birger_SN = birger_serial_number.result
+        camera_SN = camera_serial_number.result
+        lens_SN = lens_serial_number.result
+        filter_ID = filter_ID_code.result
+        serial_to_USBhub_port = serial_into_USBhub_port.result
+        camera_to_serial_port = camera_into_serial_port.result
+        USB_hub_SN = USBhub_SN.result
+        camera_to_USBhub_port = camera_into_USBhub_port.result
+
+        lens_name_dict = {'3360000099': 'name1', '3360000063': 'name2',
+                          '3360000087': 'name3', '2850000067': 'name4',
+                          '3150000110': 'name5', '5370000054': 'name6'}
+        # need to put the lens names in the above dict when they are finalised
+
+        lens_image_stabalisation = {'3360000099': True, '3360000063': True,
+                                    '3360000087': True, '2850000067': True,
+                                    '3150000110': True, '5370000054': False}
+
+        camera_dict = {'83F011167': 'sbig', '83F011791': 'sbig', '83F011639': 'sbig',
+                       '83F010801': 'sbig', '83F010774': 'sbig', '83F011758': 'sbig',
+                       '83F010771': 'sbig', '83F011810': 'sbig', 'ML5232816': 'fli', 'ML1722016': 'fli'}
+
+        additional_device = {'model': camera_dict[camera_SN],
+                             'port': camera_SN,
+                             'filter_type': filter_ID,
+                             'focuser': {'model': 'birger',
+                                         'port': birger_SN
+
+                                         },
+                             'lens': {'model': 'canon',
+                                      'port': lens_SN,
+                                      'name': lens_name_dict[lens_SN],
+                                      'image_stabalisataion': lens_image_stabalisation[lens_SN]},
+                             'USB_hub_serial_number': USB_hub_SN,
+                             'camera_into_serial_adaptor_port': camera_to_serial_port,
+                             'serial_adaptor_into_USBhub_port': serial_to_USBhub_port,
+                             'camera_into_USBhub_port': camera_to_USBhub_port
+                             }
+        if data_list['cameras']['devices'] == [None]:
+            data_list['cameras']['devices'] = [additional_device]
+        else:
+            data_list['cameras']['devices'].append(additional_device)
```

Comment:
OK, looks like the final result is a `data_list` dictionary. Maybe this should be returned so the next function can use it.

Reply:
Or maybe you need to define a `global` variable named `data_list` outside of all functions in this file, near the top of the file so all functions can access it.